crates/tabby/src/services/event.rs

```diff
@@ -64,3 +99,83 @@ impl RawEventLogger for EventService {
 pub fn create_logger() -> impl RawEventLogger {
     EventService
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn events_dir() -> PathBuf {
+        std::env::temp_dir().join(".tabby").join("events")
+    }
+
+    async fn test_event_writer_normal_write() {
+        let mut event_wr = EventWriter::new(events_dir()).await;
+        for i in 0..100 {
+            event_wr.write_line(format!("line {}", i)).await;
+        }
+        event_wr.flush().await;
+
+        let now = Utc::now();
```

Comment:
there's a rare chance this become a flacky test (e.g run at utc 23:59:59).

Is there a way to avoid it?

Reply:
that's true...nice catch!

add a helper method to fix it.

Reply:
line 129 could still break, right?

The only way to ensure it pass is to list all files writer has modified and count it in total has 100 lines?

I don't think it's actually something really useful. I'm fine to remove this test case and keep the swapping one